src/main/java/Duke.java

```diff
@@ -1,10 +1,91 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import Exception.DukeException;
+import Task.Deadline;
+import Task.Event;
+import Task.Task;
+import Task.Todo;
+
 public class Duke {
-    public static void main(String[] args) {
+    public static void main(String[] args) throws DukeException {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?\n");
+        ArrayList<Task> tasks = new ArrayList<Task>();// using ArrayList to store tasks instead of array
+        try (Scanner scan = new Scanner(System.in)) {
+            String input = scan.nextLine();
+            String output = new String();
+            while (!input.equals("bye")) {
+                try {
+                    if (input.equals("list")) {
+                        output = "Here are the tasks in the list:" + System.lineSeparator()
+                                + Task.printTasksList(tasks);
+
+                    } else if (input.startsWith("mark")) {
+                        String[] marks = input.split(" ");
+                        tasks.get(Integer.parseInt(marks[1]) - 1).markAsDone();
+                        output = "Nice! I've marked this task as done:" + System.lineSeparator()
+                                + tasks.get(Integer.parseInt(marks[1]) - 1).toString();
```

Comment:
Try extracting such chunks into a seperate method to ease readability of the main function as it seems to be too big currently